src/util.js

```diff
@@ -106,3 +107,127 @@ export function format_memory_and_limit(usage, limit) {
+                                        console.error("Failed to do GetContainerStats call:", ex, JSON.stringify(ex));
+                                        reject(new Error("Failed to do GetContainerStats call:", ex, JSON.stringify(ex)));
+                                    });
+                        });
+                        containerStatsRet.push(proEle);
+                    });
+
+                    Promise.all(containerStatsRet)
+                            .then((containerStatsRet) => {
+                                containerStatsRet.map((containerStatsRet) => {
+                                    newContainersStats[containerStatsRet.ctrId] = containerStatsRet.ctrStats;
+                                    if (newContainers.length === containerCount && Object.keys(newContainersStats).length === runCtrArr.length) {
+                                        resolve({newContainers: newContainers, newContainersStats: newContainersStats});
+                                    }
+                                });
+                            })
+                            .catch(ex => console.error("Failed to do GetContainerStats call:", ex, JSON.stringify(ex)));
+                })
+                .catch(ex => {
+                    console.error("Failed to do ListContainers call:", JSON.stringify(ex), ex.toString());
+                    reject(new Error("Failed to do ListContainers call"));
+                });
+    });
+}
+
+export function updateImages() {
+    let newImages = [];
+    let newImagesMeta = [];
+    return new Promise((resolve, reject) => {
+        varlinkCall(PODMAN, "io.podman.ListImages")
+                .then(reply => {
+                    newImagesMeta = reply.images;
+                    const imageCount = newImagesMeta ? newImagesMeta.length : 0;
+                    if (!newImagesMeta) {
+                        resolve(newImages);
+                        return;
+                    }
+                    let inspectRet = [];
+                    newImagesMeta.map((img) => {
+                        let proEle = new Promise((resolve, reject) => {
+                            varlinkCall(PODMAN, "io.podman.InspectImage", {name: img.id})
+                                    .then(reply => {
+                                        resolve(JSON.parse(reply.image));
+                                    })
+                                    .catch(ex => {
+                                        reject(new Error("Failed to do InspectImage call:", ex, JSON.stringify(ex)));
+                                    });
+                        });
+                        inspectRet.push(proEle);
+                    });
+                    Promise.all(inspectRet)
+                            .then((inspectRet) => {
+                                inspectRet.map((inspectRet) => {
+                                    newImages[inspectRet.Id] = inspectRet;
```

Comment:
Much the same comments here; array vs. dictionary conflict, please add a function comment about the return value structure, single-line function bodies, don't do the `inspectRet.push` call, etc.